applications/FluidDynamicsApplication/custom_utilities/cfd_utilities.cpp

```diff
@@ -0,0 +1,429 @@
+        }
+
+        return cell_center;
+    };
+
+    const array_1d<double, 3>& parent_center = calculate_cell_center(r_parent_geometry);
+
+    const array_1d<double, 3>& condition_center =
+        calculate_cell_center(r_condition_geometry);
+
+    return inner_prod(condition_center - parent_center, normal);
+
+    KRATOS_CATCH("");
+}
+
+void CalculateNumberOfNeighbourConditions(ModelPart& rModelPart)
+{
+    KRATOS_TRY
+
+    ModelPart::NodesContainerType& r_nodes = rModelPart.Nodes();
+
+    // reseting all nodal variables which needs to be calculated
+    VariableUtils().SetNonHistoricalVariableToZero(NUMBER_OF_NEIGHBOUR_CONDITIONS, r_nodes);
+
+    // updating
+    block_for_each(rModelPart.Conditions(), [](ConditionType& r_cond) {
+        ConditionType::GeometryType& r_geometry = r_cond.GetGeometry();
+        for (IndexType i_node = 0; i_node < r_geometry.PointsNumber(); ++i_node)
+        {
+            NodeType& r_node = r_geometry[i_node];
+            r_node.SetLock();
+            r_node.GetValue(NUMBER_OF_NEIGHBOUR_CONDITIONS) += 1;
+            r_node.UnSetLock();
+        }
+    });
+
+    rModelPart.GetCommunicator().AssembleNonHistoricalData(NUMBER_OF_NEIGHBOUR_CONDITIONS);
+
+    KRATOS_CATCH("");
+}
+
+double CalculateReactionBasedYPlusUTau(array_1d<double, 3>& rFrictionVelocity,
+                                       const array_1d<double, 3>& rReaction,
+                                       const array_1d<double, 3>& rNormal,
+                                       const double Density,
+                                       const double KinematicViscosity,
+                                       const double WallHeight)
+{
+    KRATOS_TRY
+
+    // calculating unit normal since rNormal contains surface area of the condition
+    const double surface_area = norm_2(rNormal);
+    const array_1d<double, 3>& unit_normal = rNormal / surface_area;
+
+    // calculate tangential stress
+    noalias(rFrictionVelocity) =
+        (rReaction - unit_normal * inner_prod(rReaction, unit_normal)) / surface_area;
+    const double shear_stress = norm_2(rFrictionVelocity);
+
+    // calculate y_plus
+    const double y_plus = std::sqrt(shear_stress / Density) * WallHeight / KinematicViscosity;
+
+    // calculate u_tau
+    noalias(rFrictionVelocity) =
+        rFrictionVelocity /
+        std::sqrt((shear_stress <= std::numeric_limits<double>::epsilon() ? 1.0 : shear_stress) *
+                  Density);
+
+    return y_plus;
+
+    KRATOS_CATCH("");
+}
+
+double CalculateLinearLogarithmicWallFunctionBasedYPlusLimit(const double VonKarman,
+                                                             const double WallSmoothness,
+                                                             const int MaxIterations,
+                                                             const double Tolerance)
+{
+    double y_plus = 11.06;
+    const double inv_kappa = 1.0 / VonKarman;
```

Comment:
Zero check?

Reply:
I added checks in the model part level since these methods will be called once per condition.